### PR TITLE
[login Page]Integrate Login , Signup, and logout to Firebase Auth

### PR DESCRIPTION
**Pull request by RenaufalAlrafid**
# Preview
image
image
Screenshot 2023-08-11 212405
Screenshot 2023-08-11 212430

# Task List
- Integrate Login with Firebase Auth
- Integrate Signup with Firebase Auth
- Integrate Logou with Firebase Auth

# Testing List
- [x] I have make sure that the code result is the same as in Figma
- [x] I have make sure that the code result is having dynamic size both width & height

### CODE REVIEW

File: lib/Scenes/Page/LoginPage/componnent/login_form.dart [hunk 1]
```diff
@@ -130,7 +160,10 @@ class _LoginFormState extends State<LoginForm> {
           height: 24,
         ),
         GestureDetector(
-          onTap: null,
+          onTap: () {
+            // print(true);
```
Comment (gilangpersada) [inline lib/Scenes/Page/LoginPage/componnent/login_form.dart]: dihapus aja kalo dah ga dibutuhkan

File: lib/Scenes/Page/LoginPage/componnent/signup_form.dart [hunk 2]
```diff
@@ -12,10 +14,37 @@ class _SignupFormState extends State<SignupForm> {
...
   final FocusNode _emailNode = FocusNode();
+  final TextEditingController _emailController = TextEditingController();
+  final TextEditingController _passwordController = TextEditingController();
+  final TextEditingController _usernameController = TextEditingController();
   bool _passwordFocused = false;
   bool _usernameFocused = false;
   bool _emailFocused = false;
   bool _obscureText = true;
+
+  void _HandleSignUpButton() async {
```
Comment (gilangpersada) [inline lib/Scenes/Page/LoginPage/componnent/signup_form.dart]: nama method kecil aja depannya 
```suggestion
  void _handleSignUpButton() async {
```

File: lib/Scenes/Page/LoginPage/componnent/signup_form.dart [hunk 3]
```diff
@@ -12,10 +14,37 @@ class _SignupFormState extends State<SignupForm> {
...
+    try {
+      UserCredential signupUser =
+          await FirebaseAuth.instance.createUserWithEmailAndPassword(
+        email: _emailController.text,
+        password: _passwordController.text,
+      );
+      // ignore: use_build_context_synchronously
```
Comment (gilangpersada) [inline lib/Scenes/Page/LoginPage/componnent/signup_form.dart]: ini maksudnya apa?

Comment (gilangpersada) [reply lib/Scenes/Page/LoginPage/componnent/signup_form.dart]: kasih comment yg mudah dimengerti dan ada contextnya

File: lib/Scenes/Page/LoginPage/componnent/signup_form.dart [hunk 4]
```diff
@@ -12,10 +14,37 @@ class _SignupFormState extends State<SignupForm> {
...
+      Navigator.pushReplacement(
+        context,
+        MaterialPageRoute(builder: (context) => const LoginPage()),
+      );
+    } on FirebaseAuthException catch (e) {
+      if (e.code == 'weak-password') {
```
Comment (gilangpersada) [inline lib/Scenes/Page/LoginPage/componnent/signup_form.dart]: ini harusnya ada enum nya, jangan if pake string raw value kaya gini, kalau message nya berubah nanti app kita bisa nge bug

Comment (RenaufalAlrafid) [reply lib/Scenes/Page/LoginPage/componnent/signup_form.dart]: okay lang

File: lib/Scenes/Page/LoginPage/componnent/login_form.dart [hunk 5]
```diff
@@ -32,15 +32,8 @@ class _LoginFormState extends State<LoginForm> {
...
-        setState(() {
-          _errorLogin = true;
-          _massageLogin = 'Wrong Username Or Password';
-        });
-        print('Wrong password provided for that user.');
-      }
+      print('Failed with error code: ${e.code}');
+      print(e.message);
       _errorLogin = true;
       _massageLogin = 'Wrong Username Or Password';
```
Comment (gilangpersada) [inline lib/Scenes/Page/LoginPage/componnent/login_form.dart:38]: mending pake ini pal 
`_massageLogin = e.message;`

**Review by gilangpersada — APPROVED** [review]

**Pull request merged**